### Другое

In [ ]:
import pandas as pd

"""ДОБАВИТЬ ЗАГРУЗКУ ДАННЫХ В  DataFrame"""
df = pd.DataFrame()


df_with_empty_lat = df[(df['lat'].isna()) | (df['address_rus'].isna())]
row_1 = df_with_empty_lat.iloc[0]
print(row_1)

Unnamed: 0                                         40
id                                             4510.0
atm_group                                       496.5
address        PER. BAZARNY 4            SPASSK-DALNY
address_rus                                       NaN
lat                                               NaN
long                                              NaN
target                                       0.039191
Name: 40, dtype: object


In [ ]:
#row_1 = df_with_empty_lat.head(1)['address'][0]
#print(row_1)

row_1 = 'PER. BAZARNY 4            SPASSK-DALNY'
row_1 = '14/2 BUMAZHNIY AVE MOSCOW'

from dadata import Dadata

#token = "bd862b5111b50c71399a8edd451392e4adb028de"
#secret = "d5eac715841416f41057e34da941b53282af2f6c"
token = "602bff1c26954e118781e9a109d00ab6a6de0847"
secret = "d2fa726a212448fb642337290f59605cf94f51ca"
dadata = Dadata(token, secret)
result = dadata.clean("address", row_1)
print(result)


{'source': '14/2 BUMAZHNIY AVE MOSCOW', 'result': 'г Москва, проезд Бумажный, д 14 стр 2', 'postal_code': '127015', 'country': 'Россия', 'country_iso_code': 'RU', 'federal_district': 'Центральный', 'region_fias_id': '0c5b2444-70a0-4932-980c-b4dc0d3f02b5', 'region_kladr_id': '7700000000000', 'region_iso_code': 'RU-MOW', 'region_with_type': 'г Москва', 'region_type': 'г', 'region_type_full': 'город', 'region': 'Москва', 'area_fias_id': None, 'area_kladr_id': None, 'area_with_type': None, 'area_type': None, 'area_type_full': None, 'area': None, 'city_fias_id': None, 'city_kladr_id': None, 'city_with_type': None, 'city_type': None, 'city_type_full': None, 'city': None, 'city_area': 'Северный', 'city_district_fias_id': None, 'city_district_kladr_id': None, 'city_district_with_type': 'р-н Беговой', 'city_district_type': 'р-н', 'city_district_type_full': 'район', 'city_district': 'Беговой', 'settlement_fias_id': None, 'settlement_kladr_id': None, 'settlement_with_type': None, 'settlement_type

In [ ]:
if result['metro']:
    df.at[index, 'metro'] = result['metro'][0]['distance']

In [ ]:
df.at[index, 'region_with_type'] = result['region_with_type']

##Работа с dadata

Отбор строк с пустым полем address

In [ ]:
df_address_rus_null = df[df['address_rus'].isna()]
#df_address_rus_null['metro'].head(10)
df_address_rus_null

Unnamed: 0      id  atm_group                                 address  \
40            40  4510.0      496.5  PER. BAZARNY 4            SPASSK-DALNY   
150          150   347.0     1022.0  ABB 6B NAB-CHELNINSKIJ    NAB.CHELNY     
182          182  1591.0     1942.0  16A BEREJKOVSK NAB        MOSCOW         
192          192   440.0     1942.0  113 MIRA AV.              TOGLIATTI      
214          214   556.0     1942.0  133A MOZHAYSKOE H/W.      MOSCOW         
...          ...     ...        ...                                     ...   
6158        6158  3751.0     8083.0  SEVASTOPOLSKIY ,43A.2  Magadan           
6166        6166  3794.0     8083.0  Universitetskiy 105    Volgograd         
6198        6198  4038.0     8083.0  KRONDSHTAT. ST., 113   Vyazma            
6222        6222  4209.0     8083.0  KSK 5 MKR, 29          CHita             
6247        6247  4354.0     8083.0  MOZHAYSKOE SH.,80A     Odintsovo         

     address_rus  lat  long    target area_fias_id area_with_type  ...  \
40           NaN  NaN   NaN  0.039191          NaN            NaN  ...   
150          NaN  NaN   NaN -0.145001          NaN            NaN  ...   
182          NaN  NaN   NaN  0.080034          NaN            NaN  ...   
192          NaN  NaN   NaN -0.082956          NaN            NaN  ...   
214          NaN  NaN   NaN  0.061517          NaN            NaN  ...   
...          ...  ...   ...       ...          ...            ...  ...   
6158         NaN  NaN   NaN  0.040149          NaN            NaN  ...   
6166         NaN  NaN   NaN  0.102689          NaN            NaN  ...   
6198         NaN  NaN   NaN  0.067411          NaN            NaN  ...   
6222         NaN  NaN   NaN  0.151250          NaN            NaN  ...   
6247         NaN  NaN   NaN  0.141763          NaN            NaN  ...   

     settlement_lon metro_station_name_1 metro_line_name_1  metro_distance_1  \
40              NaN                  NaN               NaN               NaN   
150             NaN                  NaN               NaN               NaN   
182             NaN                  NaN               NaN               NaN   
192             NaN                  NaN               NaN               NaN   
214             NaN                  NaN               NaN               NaN   
...             ...                  ...               ...               ...   
6158            NaN                  NaN               NaN               NaN   
6166            NaN                  NaN               NaN               NaN   
6198            NaN                  NaN               NaN               NaN   
6222            NaN                  NaN               NaN               NaN   
6247            NaN                  NaN               NaN               NaN   

     metro_station_name_2  metro_line_name_2 metro_distance_2  \
40                    NaN                NaN              NaN   
150                   NaN                NaN              NaN   
182                   NaN                NaN              NaN   
192                   NaN                NaN              NaN   
214                   NaN                NaN              NaN   
...                   ...                ...              ...   
6158                  NaN                NaN              NaN   
6166                  NaN                NaN              NaN   
6198                  NaN                NaN              NaN   
6222                  NaN                NaN              NaN   
6247                  NaN                NaN              NaN   

     metro_station_name_3 metro_line_name_3 metro_distance_3  
40                    NaN               NaN              NaN  
150                   NaN               NaN              NaN  
182                   NaN               NaN              NaN  
192                   NaN               NaN              NaN  
214                   NaN               NaN              NaN  
...                   ...               ...     

## Плохо распарсилось

In [ ]:

token = "602bff1c26954e118781e9a109d00ab6a6de0847"
secret = "d2fa726a212448fb642337290f59605cf94f51ca"
result = dadata.clean("address", '16A BEREJKOVSK NAB MOSCOW')
result

{'source': '16A BEREJKOVSK NAB MOSCOW',
 'result': 'г Москва',
 'postal_code': '101000',
 'country': 'Россия',
 'country_iso_code': 'RU',
 'federal_district': 'Центральный',
 'region_fias_id': '0c5b2444-70a0-4932-980c-b4dc0d3f02b5',
 'region_kladr_id': '7700000000000',
 'region_iso_code': 'RU-MOW',
 'region_with_type': 'г Москва',
 'region_type': 'г',
 'region_type_full': 'город',
 'region': 'Москва',
 'area_fias_id': None,
 'area_kladr_id': None,
 'area_with_type': None,
 'area_type': None,
 'area_type_full': None,
 'area': None,
 'city_fias_id': None,
 'city_kladr_id': None,
 'city_with_type': None,
 'city_type': None,
 'city_type_full': None,
 'city': None,
 'city_area': None,
 'city_district_fias_id': None,
 'city_district_kladr_id': None,
 'city_district_with_type': None,
 'city_district_type': None,
 'city_district_type_full': None,
 'city_district': None,
 'settlement_fias_id': None,
 'settlement_kladr_id': None,
 'settlement_with_type': None,
 'settlement_type': None,
 'settlem

In [ ]:
result = dadata.geolocate(name="address", lat=55.7540471, lon=37.620405)
result

[{'value': 'г Москва, Красная пл, д 9',
  'unrestricted_value': '109012, г Москва, Тверской р-н, Красная пл, д 9',
  'data': {'postal_code': '109012',
   'country': 'Россия',
   'country_iso_code': 'RU',
   'federal_district': 'Центральный',
   'region_fias_id': '0c5b2444-70a0-4932-980c-b4dc0d3f02b5',
   'region_kladr_id': '7700000000000',
   'region_iso_code': 'RU-MOW',
   'region_with_type': 'г Москва',
   'region_type': 'г',
   'region_type_full': 'город',
   'region': 'Москва',
   'area_fias_id': None,
   'area_kladr_id': None,
   'area_with_type': None,
   'area_type': None,
   'area_type_full': None,
   'area': None,
   'city_fias_id': '0c5b2444-70a0-4932-980c-b4dc0d3f02b5',
   'city_kladr_id': '7700000000000',
   'city_with_type': 'г Москва',
   'city_type': 'г',
   'city_type_full': 'город',
   'city': 'Москва',
   'city_area': 'Центральный',
   'city_district_fias_id': None,
   'city_district_kladr_id': None,
   'city_district_with_type': None,
   'city_district_type': None,
 

##Парсер

In [ ]:
from dadata import Dadata
#token = "bd862b5111b50c71399a8edd451392e4adb028de"
#secret = "d5eac715841416f41057e34da941b53282af2f6c"
token = "602bff1c26954e118781e9a109d00ab6a6de0847"
secret = "d2fa726a212448fb642337290f59605cf94f51ca"

dadata = Dadata(token, secret)

count = 0
for index, row in df_address_rus_null.iterrows():
    result = dadata.clean("address", row['address'])
    #result = dadata.suggest("address", row['address'])
    '''
    print(f"Index: {index}")
    print(f"{row}\n")
    print(row['address'])
    print(result['geo_lat'])
    '''
    count += 1
    if count >25:
      break
    df.at[index, 'area_fias_id'] = result['area_fias_id']
    df.at[index, 'area_with_type'] = result['area_with_type']
    df.at[index, 'federal_district'] = result['federal_district']
    df.at[index, 'lat'] = result['geo_lat']
    df.at[index, 'long'] = result['geo_lon']
    df.at[index, 'capital_marker'] = result['capital_marker']
    df.at[index, 'city_with_type'] = result['city_with_type']
    df.at[index, 'city_fias_id'] = result['city_fias_id']
    df.at[index, 'city_district_with_type'] = result['city_district_with_type']
    df.at[index, 'city_district_fias_id'] = result['city_district_fias_id']
    df.at[index, 'fias_id'] = result['fias_id']
    df.at[index, 'fias_level'] = result['fias_level']
    df.at[index, 'region_with_type'] = result['region_with_type']
    df.at[index, 'region_fias_id'] = result['region_fias_id']
    df.at[index, 'settlement_with_type'] = result['settlement_with_type']
    df.at[index, 'settlement_fias_id'] = result['settlement_fias_id']
    df.at[index, 'street_with_type'] = result['street_with_type']
    df.at[index, 'street_fias_id'] = result['street_fias_id']
    df.at[index, 'city_area'] = result['city_area']
    df.at[index, 'address_rus'] = result['result']
    print(row['address'], result['result'])
    if result['metro']:
      df.at[index, 'metro'] = result['metro'][0]['distance']
    #print(count)

    '''
    Нет данных 'city_lat', 'city_lon', 'region_lat', 'region_lon', 'street_lat', 'street_lon'
     'area_lat', 'area_lon', 'settlement_lat',
       'settlement_lon', 'metro_station_name_1', 'metro_line_name_1',
       'metro_distance_1', 'metro_station_name_2', 'metro_line_name_2',
       'metro_distance_2', 'metro_station_name_3', 'metro_line_name_3',
       'metro_distance_3'

     'area_fias_id', 'area_with_type', 'city_with_type',
       'city_fias_id', 'federal_district', 'capital_marker', 'fias_id',
       'fias_level', 'region_with_type', 'region_fias_id',
       'settlement_with_type', 'settlement_fias_id', 'street_with_type',
       'street_fias_id', 'city_lat', 'city_lon', 'region_lat', 'region_lon',
       'street_lat', 'street_lon', 'city_area', 'city_district_fias_id',
       'city_district_with_type', 'area_lat', 'area_lon', 'settlement_lat',
       'settlement_lon', 'metro_station_name_1', 'metro_line_name_1',
       'metro_distance_1', 'metro_station_name_2', 'metro_line_name_2',
       'metro_distance_2', 'metro_station_name_3', 'metro_line_name_3',
       'metro_distance_3'],
     '''

    # Add more fields as needed

PER. BAZARNY 4            SPASSK-DALNY Приморский край, г Спасск-Дальний, пер Базарный, д 4
ABB 6B NAB-CHELNINSKIJ    NAB.CHELNY   Респ Татарстан, г Набережные Челны
16A BEREJKOVSK NAB        MOSCOW       г Москва
113 MIRA AV.              TOGLIATTI    Самарская обл, г Тольятти, ул Мира, влд 113
133A MOZHAYSKOE H/W.      MOSCOW       г Москва, ш Можайское, д 133А
63A IZMAYLOVSKIY AV.      MOSCOW       г Москва, ул Измайловский Вал, д 63А
14/2 BUMAZHNIY AVE        MOSCOW       г Москва, проезд Бумажный, д 14 стр 2
45 YANA RAYNISA ST.       MOSCOW       г Москва, б-р Яна Райниса, д 45
45 YANA RAYNISA ST.       MOSCOW       г Москва, б-р Яна Райниса, д 45
45 YANA RAYNISA ST.       MOSCOW       г Москва, б-р Яна Райниса, д 45
45 YANA RAYNISA ST.       MOSCOW       г Москва, б-р Яна Райниса, д 45
8 NOVATOROV STR.          S.-PETERBURG г Санкт-Петербург, б-р Новаторов, д 8
199 G.IM.KRASNOYARSKIY RA KRASNOYARSK  г Красноярск, пр-кт им.газеты "Красноярский рабочий", д 199
107 UNIVERSITETSKIY  

Приморский край, г Спасск-Дальний, пер Базарный, д 4
Респ Татарстан, г Набережные Челны
г Москва
Самарская обл, г Тольятти, ул Мира, влд 113
г Москва, ш Можайское, д 133А
г Москва, ул Измайловский Вал, д 63А
г Москва, проезд Бумажный, д 14 стр 2
г Москва, б-р Яна Райниса, д 45
г Москва, б-р Яна Райниса, д 45
г Москва, б-р Яна Райниса, д 45
г Москва, б-р Яна Райниса, д 45
г Санкт-Петербург, б-р Новаторов, д 8
г Красноярск, пр-кт им.газеты "Красноярский рабочий", д 199
г Волгоград, пр-кт Университетский, д 107
г Кемерово, б-р Строителей, д 28
Самарская обл, г Сызрань, ш Ульяновское, д 38
г Москва, наб Семеновская, д 16Б
г Томск, пл Ленина, д 317, кв 2
г Саратов, ул Большая Садовая, д 153/163
г Москва, наб Пресненская
г Москва, пл Комсомольская, д 2
г Омск, ул Российская, д 6
г Санкт-Петербург, пл Комендантская
Алтайский край, г Бийск, тер. СНТ Заря, ул N 1, д 23
г Новосибирск, ул Гоголя
г Архангельск, пр-кт Обводный канал, д 91
г Ульяновск, пр-кт Ульяновский, д 6
г Санкт-Петербург
г Брянск, ул 50-й Армии, д 1
г Пермь, пр-кт Парковый, д 66
Иркутская обл, г Усолье-Сибирское
г Улан-Удэ, ул Коммунистическая, д 47А
г Улан-Удэ, ул Коммунистическая, д 47А
г Барнаул, пр-кт Красноармейский, д 47А
г Улан-Удэ, ул Коммунистическая, д 47А
г Улан-Удэ, ул Коммунистическая, д 47А
Ханты-Мансийский Автономный округ - Югра АО, г Сургут, б-р Свободы, д 2 к 2
Новгородская обл, Шимский р-н, д Нижний Прихон, д 22Б
г Нижний Новгород, ул Максима Горького, д 262
г Кемерово, пр-кт Ленинградский, д 29В
г Улан-Удэ, ул Терешковой, д 26А
г Краснодар, ул Красноармейская, д 116/2
г Иркутск, мкр Юбилейный, д 19/1
г Томск, пр-кт Комсомольский, д 13Б
г Кострома, пл Широкова В.Ф.
г Ростов-на-Дону, пр-кт Ленина, д 56
г Ростов-на-Дону, пр-кт Ленина, д 56
г Москва, ул Хуторская 2-я, д 38А стр 9
г Москва, проезд 1-й Красногвардейский, д 15
г Москва, б-р Дмитрия Донского

In [ ]:
df

Unnamed: 0      id  atm_group                                 address  \
0              0  8526.0       32.0  EMELYANOVA,34          Y-SAKHALINSK      
1              1  8532.0       32.0  KOMSOMOLSKAYA,259B     Y.SAKHALINSK      
2              2  8533.0       32.0  KOMMUN. PR., 32        YUZHNO SAKHAL     
3              3  8684.0       32.0  LENINGRADSKIY PR.,76A  MOSCOW            
4              4    37.0       32.0  GVARDEYSKAYA PL., 2    NORILSK           
...          ...     ...        ...                                     ...   
6256        6256  4402.0     8083.0  FEDORA POPOVA, 14      Yakutsk           
6257        6257  4421.0     8083.0  STROITELEY, 43         Novokuznetsk      
6258        6258  4422.0     8083.0  Rublevskoye sh, 62     Moskva            
6259        6259  4426.0     8083.0  EFANOVA, 22            Zyryanskoye       
6260        6260  4429.0     8083.0  OMMUNISTICHESKAYA,10A  Kizhinga          

                                            address_rus        lat  \
0     улица А.О. Емельянова, 34, Южно-Сахалинск, Сах...  46.940995   
1     Комсомольская улица, 259, Южно-Сахалинск, Саха...  46.937353   
2     Коммунистический проспект, Южно-Сахалинск, Сах...  46.959413   
3     Ленинградский проспект, 76А, Москва, Россия, 1...  55.805827   
4     Гвардейская площадь, 2, Норильск, Красноярский...  69.343541   
...                                                 ...        ...   
6256  улица Федора Попова, 14, Якутск, Республика Са...  62.045259   
6257  проспект Строителей, 43, Новокузнецк, Кемеровс...  53.769459   
6258       Рублёвское шоссе, 62, Москва, Россия, 121609  55.766212   
6259  улица Ефанова, 22, село Зырянское, Зырянский р...  56.838223   
6260  Коммунистическая улица, 10А, село Кижинга, Киж...  51.848725   

            long    target                          area_fias_id  \
0     142.738319  0.011500                                   NaN   
1     142.753348  0.029710                                   NaN   
2     142.741113  0.009540                                   NaN   
3      37.515146 -0.094035                                   NaN   
4      88.211228  0.079277                                   NaN   
...          ...       ...                                   ...   
6256  129.748896  0.207916                                   NaN   
6257   87.117251  0.062220                                   NaN   
6258   37.380875  0.108743                                   NaN   
6259   86.640829  0.148938  9e3d4605-701e-4733-b6ed-1d03a2b5d4fe   
6260  109.905102  0.089919  d142ee10-5d45-4de9-87e6-74b3aa7fe178   

        area_with_type  ... metro_station_name_1    metro_line_name_1  \
0                  NaN  ...                  NaN                  NaN   
1                  NaN  ...                  NaN                  NaN   
2                  NaN  ...                  NaN                  NaN   
3                  NaN  ...                Сокол       Замоскворецкая   
4                  NaN  ...                  NaN                  NaN   
...                ...  ...                  ...                  ...   
6256               NaN  ...                  NaN                  NaN   
6257               NaN  ...                  NaN                  NaN   
6258               NaN  ...           Крылатское  Арбатско-Покровская   
6259     Зырянский р-н  ...                  NaN                  NaN   
6260  Кижингинский р-н  ...                  NaN                  NaN   

     metro_distance_1 metro_station_name_2    metro_line_name_2  \
0                 NaN                  NaN                  NaN   
1                 NaN                  NaN                  NaN   
2                 NaN                  NaN                  NaN   
3                 0.0             Аэропорт       Замоскворецкая   
4                 NaN                  NaN                  NaN   
...               ...                  ...                  ...   
6256              NaN                  NaN           

In [ ]:
#Пропущенные значения
df.isna().sum()

Unnamed: 0                    0
id                            0
atm_group                     0
address                       0
address_rus                 284
lat                         284
long                        284
target                        0
area_fias_id               5441
area_with_type             5441
city_with_type              626
city_fias_id                626
federal_district            327
capital_marker              327
fias_id                     327
fias_level                  327
region_with_type            327
region_fias_id              327
settlement_with_type       5768
settlement_fias_id         5768
street_with_type            513
street_fias_id              514
city_lat                    635
city_lon                    635
region_lat                 5498
region_lon                 5498
street_lat                  532
street_lon                  532
city_area                  5708
city_district_fias_id      6131
city_district_with_type    4511
area_lat

In [ ]:
df_address_rus_null.head(20)

Unnamed: 0      id  atm_group                                 address  \
40           40  4510.0      496.5  PER. BAZARNY 4            SPASSK-DALNY   
150         150   347.0     1022.0  ABB 6B NAB-CHELNINSKIJ    NAB.CHELNY     
182         182  1591.0     1942.0  16A BEREJKOVSK NAB        MOSCOW         
192         192   440.0     1942.0  113 MIRA AV.              TOGLIATTI      
214         214   556.0     1942.0  133A MOZHAYSKOE H/W.      MOSCOW         
216         216   568.0     1942.0  63A IZMAYLOVSKIY AV.      MOSCOW         
218         218   587.0     1942.0  14/2 BUMAZHNIY AVE        MOSCOW         
232         232   705.0     1942.0  45 YANA RAYNISA ST.       MOSCOW         
233         233   706.0     1942.0  45 YANA RAYNISA ST.       MOSCOW         
234         234   707.0     1942.0  45 YANA RAYNISA ST.       MOSCOW         
235         235   708.0     1942.0  45 YANA RAYNISA ST.       MOSCOW         
255         255  1680.0     1942.0  8 NOVATOROV STR.          S.-PETERBURG   
261         261  1712.0     1942.0  199 G.IM.KRASNOYARSKIY RA KRASNOYARSK    
324         324  1307.0     1942.0  107 UNIVERSITETSKIY       VOLGOGRAD      
327         327  1338.0     1942.0  28 STROITELEY BLVD        KEMEROVO       
334         334  1380.0     1942.0  38 ULYANOVSKOE H/W        SYZRAN         
336         336  1384.0     1942.0  16 B.SEMENOVSKAYA         MOSCOW         
337         337  1389.0     1942.0  317-2 LENINA AV.          TOMSK          
359         359  1577.0     1942.0  153/163 B.SADOVAYA STR    SARATOV        
440         440  2491.0     3185.5  PRESNENSKAJA              MOSKVA         

    address_rus         lat         long    target area_fias_id  \
40          NaN  44.5901175  132.8157288  0.039191         None   
150         NaN  55.7434619   52.3959165 -0.145001         None   
182         NaN  55.7540471    37.620405  0.080034         None   
192         NaN  53.5107844   49.4444678 -0.082956         None   
214         NaN  55.7183935   37.4151538  0.061517         None   
216         NaN   55.784549   37.7178968  0.008276         None   
218         NaN  55.7896669   37.5869132 -0.052220         None   
232         NaN  55.8500783   37.4066105  0.097726         None   
233         NaN  55.8500783   37.4066105  0.067342         None   
234         NaN  55.8500783   37.4066105  0.162231         None   
235         NaN  55.8500783   37.4066105  0.128152         None   
255         NaN  59.8505553   30.2693604 -0.010433         None   
261         NaN  55.9907694   92.8878625  0.048500         None   
324         NaN  48.6364995   44.4360268  0.044479         None   
327         NaN  55.3447661   86.1674323  0.033266         None   
334         NaN  53.1803506   48.4908215  0.053041         None   
336         NaN   55.776462    37.702108  0.095677         None   
337         NaN  56.4868178   84.9472304 -0.071347         None   
359         NaN  51.5414412   45.9900106 -0.078090         None   
440         NaN  55.7471444   37.5419492  0.006006         None   

    area_with_type  ... metro_station_name_1 metro_line_name_1  \
40            None  ...                  NaN               NaN   
150           None  ...                  NaN               NaN   
182           None  ...                  NaN               NaN   
192           None  ...                  NaN               NaN   
214           None  ...                  NaN               NaN   
216           None  ...                  NaN               NaN   
218           None  ...                  NaN               NaN   
232           None  ...                  NaN               NaN   
233           None  ...                  NaN               NaN   
234           None  ...                  NaN               NaN   
235           None  ...                  NaN               NaN   
255           None  ...                  NaN               NaN   
261           None  ...                  NaN               NaN   
324           None  ...           

In [ ]:
df[df['address_rus'].isna()].head(20)

Unnamed: 0      id  atm_group                                 address  \
451         451  2557.0     3185.5  KOMSOMOLSKAJA P.,2        MOSKVA         
475         475  2746.0     3185.5  KR.G. ROSSIJSKAJA,6       OMSK           
498         498  8413.0     5478.0  PL. KOMENDANTSKAYA        SANKT-PETERB   
506         506  8436.0     5478.0  D. 23, KORP. 1, UL. MATRO BIYSK G        
534         534  4871.0     5478.0  UL. GOGOLYA               NOVOSIBIRSK    
566         566  5021.0     5478.0  D. 91, PR-KT OBVODNYI KAN ARKHANGELSK    
587         587  5169.0     5478.0  D. 6, PR-KT UL'YANOVSKII  ULYANOVSK G    
602         602  5284.0     5478.0  D. 12, LIT. V, UL. FAVORS SANKT-PETERB   
603         603  5304.0     5478.0  D. 1, UL. 50-I ARMII      BRYANSK G      
617         617  5391.0     5478.0  D. 66, PR-KT PARKOVYI     PERM G         
641         641  5581.0     5478.0  79, UL. INTERNACIONAL'NAY USOLYE-SIBIR   
675         675  5877.0     5478.0  D. 47 A, UL. KOMMUNISTICH ULAN-UDE G     
676         676  5878.0     5478.0  D. 47 A, UL. KOMMUNISTICH ULAN-UDE G     
700         700  5980.0     5478.0  D. 47A, PR-KT KRASNOARMEI BARNAUL G      
710         710  6075.0     5478.0  D. 47 A, UL. KOMMUNISTICH ULAN-UDE G     
711         711  6076.0     5478.0  D. 47 A, UL. KOMMUNISTICH ULAN-UDE G     
751         751  6404.0     5478.0  D. 2, KORP. 2, B-R SVOBOD SURGUT G       
761         761  6464.0     5478.0  D. 22B, PR-KT KORABLESTRO NIZHNIY NOVG   
780         780  6576.0     5478.0  D. 262, UL. MAKSIMA GOR'K NIZHNIY NOVG   
877         877  7249.0     5478.0  D. 29V, PR-KT LENINGRADSK KEMEROVO G     

    address_rus  lat long    target area_fias_id area_with_type  ...  \
451         NaN  NaN  NaN -0.071913          NaN            NaN  ...   
475         NaN  NaN  NaN -0.122659          NaN            NaN  ...   
498         NaN  NaN  NaN -0.037399          NaN            NaN  ...   
506         NaN  NaN  NaN -0.137240          NaN            NaN  ...   
534         NaN  NaN  NaN -0.057024          NaN            NaN  ...   
566         NaN  NaN  NaN -0.069187          NaN            NaN  ...   
587         NaN  NaN  NaN -0.010810          NaN            NaN  ...   
602         NaN  NaN  NaN -0.050785          NaN            NaN  ...   
603         NaN  NaN  NaN -0.046154          NaN            NaN  ...   
617         NaN  NaN  NaN -0.054458          NaN            NaN  ...   
641         NaN  NaN  NaN -0.080696          NaN            NaN  ...   
675         NaN  NaN  NaN  0.004058          NaN            NaN  ...   
676         NaN  NaN  NaN  0.003823          NaN            NaN  ...   
700         NaN  NaN  NaN  0.015838          NaN            NaN  ...   
710         NaN  NaN  NaN -0.018201          NaN            NaN  ...   
711         NaN  NaN  NaN  0.022241          NaN            NaN  ...   
751         NaN  NaN  NaN -0.028232          NaN            NaN  ...   
761         NaN  NaN  NaN -0.019509          NaN            NaN  ...   
780         NaN  NaN  NaN -0.073875          NaN            NaN  ...   
877         NaN  NaN  NaN -0.025233          NaN            NaN  ...   

    metro_station_name_1 metro_line_name_1 metro_distance_1  \
451                  NaN               NaN              NaN   
475                  NaN               NaN              NaN   
498                  NaN               NaN              NaN   
506                  NaN               NaN              NaN   
534                  NaN               NaN              NaN   
566                  NaN               NaN              NaN   
587                  NaN               NaN              NaN   
602                  NaN               NaN              NaN   
603                  NaN               NaN              NaN   
617                  NaN               NaN              NaN   
641                  NaN               NaN              NaN   
675                  NaN               NaN              NaN   
676                  NaN          

## Транслитерация

In [1]:
# https://github.com/hflabs/dadata-py
from dadata import Dadata
from transliterate import translit
token = "bd862b5111b50c71399a8edd451392e4adb028de"
secret = "d5eac715841416f41057e34da941b53282af2f6c"
dadata = Dadata(token, secret)

row_1 = df.head(1)['address'][0]
text = translit(row_1, 'ru')
print(row_1, text)

#result = dadata.clean("address", row_1)
result = dadata.suggest("address", row_1)
print(result)
#print(type(result[1]),result[1])

#print(result[1]['data']['region_with_type'])
#print(result[1]['data']['geo_lat'])

ModuleNotFoundError: No module named 'transliterate'

In [ ]:
df.head(1)['address_rus'][0]

'улица А.О. Емельянова, 34, Южно-Сахалинск, Сахалинская область, Россия'

In [ ]:
from pprint import pprint
from dadata import Dadata

token = "bd862b5111b50c71399a8edd451392e4adb028de"
dadata = Dadata(token)
result = dadata.geolocate(name="address", lat=df.iloc[0]['lat'], lon=df.iloc[0]['long'])
pprint(result)
print(type(result))

[{'data': {'area': None,
           'area_fias_id': None,
           'area_kladr_id': None,
           'area_type': None,
           'area_type_full': None,
           'area_with_type': None,
           'beltway_distance': None,
           'beltway_hit': None,
           'block': None,
           'block_type': None,
           'block_type_full': None,
           'capital_marker': '2',
           'city': 'Южно-Сахалинск',
           'city_area': None,
           'city_district': None,
           'city_district_fias_id': None,
           'city_district_kladr_id': None,
           'city_district_type': None,
           'city_district_type_full': None,
           'city_district_with_type': None,
           'city_fias_id': '44388ad0-06aa-49b0-bbf9-1704629d1d68',
           'city_kladr_id': '6500000100000',
           'city_type': 'г',
           'city_type_full': 'город',
           'city_with_type': 'г Южно-Сахалинск',
           'country': 'Россия',
           'country_iso_code': 'RU',
  

In [ ]:
len(result)


7

In [ ]:
df.iloc[0]

Unnamed: 0                                                     0
id                                                        8526.0
atm_group                                                   32.0
address                   EMELYANOVA,34          Y-SAKHALINSK   
address_rus    улица А.О. Емельянова, 34, Южно-Сахалинск, Сах...
lat                                                    46.940995
long                                                  142.738319
target                                                    0.0115
Name: 0, dtype: object

In [ ]:
for obj in result:
  print(str(obj['data']['street']) + str(obj['data']['house']))

Емельянова А.О.34
Емельянова А.О.34А
Емельянова А.О.34Г
Емельянова А.О.31/1
Емельянова А.О.36
Емельянова А.О.32
NoneNone


In [ ]:
result[1]['data']['street'] + result[1]['data']['house']

'Емельянова А.О.34А'